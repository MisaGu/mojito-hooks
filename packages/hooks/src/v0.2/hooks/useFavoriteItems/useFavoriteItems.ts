import { EMojitoQueries } from '../../domain/gql/queries';
import { IMojitoFavoriteResponse } from '../../domain/interfaces';
import { BaseQueryHookProps } from '../../domain/interfaces/hooks.interface';
import { useMojitoFactory } from '../useMojitoFactory/useMojitoFactory';

function transformFn(response?: IMojitoFavoriteResponse | null) {
  if (!response) return undefined;

  return response.me.favoriteItems;
}

export type UseFavoriteItemsData = ReturnType<typeof transformFn>;

export type UseFavoriteItemsReturn = ReturnType<typeof useFavoriteItems>;

export type UseFavoriteItemsProps = BaseQueryHookProps<UseFavoriteItemsData>;

export function useFavoriteItems({ options }: UseFavoriteItemsProps = {}) {
  return useMojitoFactory({
    as: 'favoriteItems',
    query: EMojitoQueries.userFavorites,
    options,
    transformFn,
  });
}
