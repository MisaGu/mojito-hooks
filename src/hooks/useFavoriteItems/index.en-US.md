---
nav:
  path: /hooks
  group: Mojito
---

# useFavoriteItems

Returns the current user `id` and all its favorite items.

## Examples

<code src="./demo/demo1.tsx" />
