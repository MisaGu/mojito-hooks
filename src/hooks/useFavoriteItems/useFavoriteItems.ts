import { QueryKey } from '../../domain/utils/queryKeyFactory.util';
import { EMojitoKey } from '../../domain/enums/state.enum';
import { FavoriteResponse } from '../../domain/interfaces';
import { BaseQueryHookProps } from '../../domain/interfaces/hooks.interface';
import { useMojitoFactory } from '../useMojitoFactory/useMojitoFactory';

function selectorFn(response?: FavoriteResponse) {
  if (!response) return undefined;

  return response.me.favoriteItems;
}

export type UseFavoriteItemsData = ReturnType<typeof selectorFn>;

export type UseFavoriteItemsReturn = ReturnType<typeof useFavoriteItems>;

export type UseFavoriteItemsProps = BaseQueryHookProps<UseFavoriteItemsData>;

export function useFavoriteItems({ options }: UseFavoriteItemsProps = {}) {
  return useMojitoFactory({
    as: 'favoriteItems',
    queryKey: QueryKey.get(EMojitoKey.userFavorites),
    options,
    selectorFn,
    onlyAuthenticated: true,
  });
}

export default useFavoriteItems;
