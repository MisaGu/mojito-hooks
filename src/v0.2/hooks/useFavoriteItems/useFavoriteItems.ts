import { EMojitoKey } from '../../domain/enums/state.enum';
import { FavoriteResponse } from '../../domain/interfaces';
import { BaseQueryHookProps } from '../../domain/interfaces/hooks.interface';
import { useMojitoFactory } from '../useMojitoFactory/useMojitoFactory';

function selectorFn(response?: FavoriteResponse) {
  if (!response) return undefined;

  return response.me.favoriteItems;
}

export type UseFavoriteItemsData = ReturnType<typeof selectorFn>;

export type UseFavoriteItemsReturn = ReturnType<typeof useFavoriteItems>;

export type UseFavoriteItemsProps = BaseQueryHookProps<UseFavoriteItemsData>;

export function useFavoriteItems({ options }: UseFavoriteItemsProps = {}) {
  return useMojitoFactory({
    as: 'favoriteItems',
    query: EMojitoKey.userFavorites,
    options,
    selectorFn,
    onlyAuthenticated: true,
  });
}

export default useFavoriteItems;
