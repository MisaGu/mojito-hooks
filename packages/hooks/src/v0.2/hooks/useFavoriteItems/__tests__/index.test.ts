import { useFavoriteItems } from '../useFavoriteItems';

describe('useToggle', () => {
  it('should be defined', () => {
    expect(useFavoriteItems).toBeDefined();
  });
});
