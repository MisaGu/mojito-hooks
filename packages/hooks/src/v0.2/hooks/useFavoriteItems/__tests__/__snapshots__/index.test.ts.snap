// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useFavoriteItems() returns the right result when needed 1`] = `
Object {
  "error": null,
  "favoriteItems": Array [],
  "isLoading": false,
  "queryResult": Object {
    "dataUpdatedAt": Any<Number>,
    "errorUpdateCount": 0,
    "errorUpdatedAt": 0,
    "failureCount": 0,
    "isError": false,
    "isFetched": true,
    "isFetchedAfterMount": true,
    "isFetching": false,
    "isIdle": false,
    "isLoadingError": false,
    "isPlaceholderData": false,
    "isPreviousData": false,
    "isRefetchError": false,
    "isRefetching": false,
    "isStale": false,
    "isSuccess": true,
    "remove": [Function],
    "status": "success",
  },
  "refetch": [Function],
}
`;
